### DISCUSSION TITLE
IKEv1: invalid HASH_V1 payload length, Payload Malformed

### DISCUSSION THREAD

**Discussion by emanvanhoot** [strongswan/strongswan#600] [Q&A] [answered]
Hi,

I'm trying to get Strongswan to set up a site-site VPN via IKEv1 using PSK, between two versions of SS: 5.8.4 (client/responder) and 4.6.4 (initiator). I have configuration control of 5.8.4, and access to 4.6.4, but I cannot change the configuration of the older version.

I keep getting:

```
06[ENC] invalid HASH_V1 payload length, decryption failed?
06[ENC] could not decrypt payloads
06[IKE] message parsing failed 
06[ENC] generating INFORMATIONAL_V1 request 4287511576 [ HASH N(PLD_MAL) ] 
```

on the responder.  The shared secret is trivial, and there is only one.  I know this error often suggests the shared secret is the problem, but I can't see how that could be.  If I move both ends to IKEv2 with an almost identical config (and same shared secret), then I can set up the tunnel without issue.  The network config seems right for that reason, and its really pretty basic on top of it.  The ike/esp proposals all check out and everyone seems happy, and the CHILD SA's appear to be found/agreed-upon.  We just stop being able to communicate with the older SS device. 

Here is a look at the network.  Networks on each end (192.168.1.0/24, 192.168.2.0/24), and connecting interfaces on either end of the SS gateways (the bracketed guys) are both on 182.168.1.0/24.  These are all private networks at the moment, and not routed externally:

`
192.168.1.0/24 <---> 192.168.1.42 [SS 5.8.4] 182.168.1.42 <---> 182.168.1.22 [SS 4.6.4]192.168.2.22 <---> 192.168.2.0/24
`

Is there something besides the PSK that could be causing this?  Is there some incompatibility between the versions of SS?  I've attached the logs/etc as well.

Thanks so much for the help.
[strongswan_configs.tar.gz](https://github.com/strongswan/strongswan/files/7180579/strongswan_configs.tar.gz)

**Comment by tobiasbrunner** [comment, marked as answer]
> and 4.6.4 (initiator)

You shouldn't use such an old version anymore (and IKEv1 neither).

> Is there something besides the PSK that could be causing this?

It's definitely not the PSK during Quick Mode. The problem is that the daemon only keeps track of 3 concurrent Quick Mode exchanges. If there are more, IVs for previous exchanges are dropped and messages can't be decrypted (a Quick Mode exchange consists of three messages). Try increasing `charon.max_ikev1_exchanges` in strongswan.conf.

**Comment by emanvanhoot** [comment]
Thanks for the reply, Tobias. 

That fixed 5.8.4 as the responder (different issue when it initiates), and I would have never thought of that as the solution.

Thanks!
